stdlib/public/Distributed/DistributedActorSystem.swift

```diff
@@ -333,38 +351,26 @@ public protocol DistributedTargetInvocation {
   associatedtype SerializationRequirement
 
   // === Sending / recording  -------------------------------------------------
-
-  // RECIPIENT:
-  // when we get the gen args; we can check them against the where clause
-  // and throw, rather than blow up... we have the ability to check the where clause
-  // the same code as implements `as?` takes generic signature and checks type arguments against that
-  //
-  // accessor is not generic, gets the subs passed in, forms SubsMap and uses to look up other things
-  // pass to call emission which would do the right thing...
-  /// Ad-hoc requirement
-  ///
   /// The arguments must be encoded order-preserving, and once `decodeGenericSubstitutions`
   /// is called, the substitutions must be returned in the same order in which they were recorded.
-  mutating func recordGenericSubstitution<T>(mangledType: T.Type) throws
+  mutating func recordGenericSubstitution<T>(_ type: T.Type) throws
 
 //  /// Ad-hoc requirement
 //  ///
 //  /// Record an argument of `Argument` type in this arguments storage.
 //  mutating func recordArgument<Argument: SerializationRequirement>(argument: Argument) throws
+  mutating func recordErrorType<E: Error>(_ type: E.Type) throws
 
 //  /// Ad-hoc requirement
 //  ///
-//  mutating func recordReturnType<R: SerializationRequirement>(mangledType: R.Type) throws
-
-  mutating func recordErrorType<E: Error>(mangledType: E.Type) throws
-
+//  /// Record the return type of the distributed method.
+//  mutating func recordReturnType<R: SerializationRequirement>(_ type: R.Type) throws
   mutating func doneRecording() throws
 
   // === Receiving / decoding -------------------------------------------------
-
   mutating func decodeGenericSubstitutions() throws -> [Any.Type]
 
-  mutating func argumentDecoder() -> Self.ArgumentDecoder
+  func makeArgumentDecoder() -> Self.ArgumentDecoder
 
   mutating func decodeReturnType() throws -> Any.Type?
```

Comment:
`record*` methods have to be `mutating` but why would `decode*` have to?

Reply:
good point, they dont :)